App/Parameter-Parser/parameter_parser.go

```diff
@@ -0,0 +1,290 @@
+	var parsedParams []commandParameter
+	var currentParam string
+	var args []string
+	argMode := false
+	for i := 2; i < userParametersLen; i++ {
+		token := userParameters[i]
+		v, ok := params[token]
+		if ok {
+			if argMode {
+				argMode = false
+				if len(args) == 0 {
+					panic(parsingError{
+						Code: 303,
+						Message: `Parsing error occurred. This could be due to:
+							- too few arguments passed to arg required param`,
+					})
+				}
+				checkOccurences(args)
+				b := params[currentParam].ArgCount
+				if b == 1 {
+					if len(args) != b {
+						panic(parsingError{
+							Code: 306,
+							Message: `Parsing error occurred. This could be due to:
+								- unnecessary argument passed to the parameter`,
+						})
+					}
+				}
+				argCopy := append([]string(nil), args...)
+				parsedParams = append(parsedParams, commandParameter{
+					Name:      currentParam,
+					Arguments: argCopy,
+				})
+				// clear args for reuse
+				args = args[:0]
+			}
+			if v.ArgRequired {
+				if userParametersLen == i+1 {
+					panic(parsingError{
+						Code: 303,
+						Message: `Parsing error occurred. This could be due to:	
+							- too few arguments passed to arg required param`,
+					})
+				}
+				argMode = true
+				currentParam = token
+			} else {
+				if userParametersLen > i+1 {
+					next := userParameters[i+1]
+					_, ok := params[next]
+					if ok {
+						parsedParams = append(parsedParams, commandParameter{
+							Name:      token,
+							Arguments: []string{v.DefaultVal},
+						})
+						continue
+					} else {
+						parsedParams = append(parsedParams, commandParameter{
+							Name:      token,
+							Arguments: []string{next},
+						})
+						i++
+						continue
+					}
+				} else {
+					parsedParams = append(parsedParams, commandParameter{
+						Name:      token,
+						Arguments: []string{v.DefaultVal},
+					})
+					continue
+				}
+			}
+		} else {
+			if argMode {
+				v, _ := params[currentParam]
+				if len(v.Arguments) > 0 {
+					if !findElement(token, v.Arguments) {
+						panic(parsingError{
+							Code: 304,
+							Message: `Parsing error occurred. This could be due to:
+								- invalid argument passed to the parameter`,
+						})
+					}
+					args = append(args, token)
+				} else {
+					args = append(args, token)
+				}
+			} else {
+				panic(parsingError{
+					Code: 304,
+					Message: `Parsing error occurred. This could be due to:
+						- invalid argument passed to the parameter`,
+				})
+			}
+		}
+	}
+	if argMode {
+		argMode = false
+		checkOccurences(args)
+		argCopy := append([]string(nil), args...)
+		parsedParams = append(parsedParams, commandParameter{
+			Name:      currentParam,
+			Arguments: argCopy,
+		})
+		args = args[:0]
+	}
+
+	return parsedParams
+}
+func findElement(userParam string, params []string) bool {
+	for i := 0; i < len(params); i++ {
+		if params[i] == userParam {
+			return true
+		}
+	}
+	return false
+}
+func checkOccurences(args []string) {
```

Comment:
Zmienić nazwę tej funkcji bo będzie krzyczał za każdym razem.
```suggestion
func checkOccurrences(args []string) {
```